Doc/library/test.rst

```diff
@@ -1087,12 +1087,12 @@ The :mod:`test.support.socket_helper` module provides support for socket tests.
    buildbot environment.  This method raises an exception if the
    ``sock.family`` is :const:`~socket.AF_INET` and ``sock.type`` is
    :const:`~socket.SOCK_STREAM`, and the socket has
-   :const:`~socket.SO_REUSEADDR` or :const:`~socket.SO_REUSEPORT` set on it.
+   :const:`!SO_REUSEADDR` or :const:`!SO_REUSEPORT` set on it.
```

Comment:
The better way to fix these warnings would be to add documentation for these constants in the `socket` module docs, if that documentation doesn't already exist

Reply:
Perhaps, but ISTR the `socket` module doc explicitly mentions a whole raft of constants as coming from the underlying C socket(2) implementation. That is presumably to keep from duplicating underlying canonical documentation and to avoid introducing mistakes (think platform dependencies).

I understand it would be great to have the Python library docs completely self-contained, but I don't think that's the best way to go in some cases.

Reply:
Fair point, there are probably too many in the socket module for us to list them all out explicitly!

Reply:
Given that many of the possible constants are only given in the `socket` module doc as wildcards (e.g., `SO_*`) I don't think that will work unless you only want to list a known (small) subset.

Reply:
This stuff is really at the fringe between low-level Unix stuff and the relevant Python wrappers. According to Wikipedia, the first edition of Stevens' _Unix Network Programming_ (the network programming bible when I was a wee lad) is 768 pages long. It's not at all clear to me that there's much to be gained by documenting a few socket options. It's not really going to flatten the socket programming learning curve much. Programmers will have to delve into the C socket documentation to do anything serious anyway.

Sorry about the soapbox.